Google.Api.Gax.Grpc/CallSettingsExtensions.cs

```diff
@@ -82,16 +73,8 @@ public static CallSettings MergedWith(this CallSettings original, CallSettings o
         /// not present in the new call settings. If both this and <paramref name="settings"/> are null,
         /// the return value is null.</param>
         /// <returns>A new set of call settings, or null if both parameters are null.</returns>
-        public static CallSettings WithRetry(
-            this CallSettings settings,
-            RetrySettings retry) =>
-            settings == null
-                ? CallSettings.FromRetry(retry)
-                // But as long as user code can specify credentials we should continue to respect that.
-                : new CallSettings(settings.CancellationToken,
-                    settings.Expiration, retry, settings.HeaderMutation,
-                    settings.WriteOptions, settings.PropagationToken,
-                    settings.ResponseMetadataHandler, settings.TrailingMetadataHandler);
+        public static CallSettings WithRetry(this CallSettings settings, RetrySettings retry) =>
```

Comment:
This isn't actually the same thing, as I found out today - with the old code, you could call WithRetry(null) and it would return a CallSettings with no retry. This code won't do that, as CallSettings.FromRetry(null) returns null.

Ditto for the expiration, I suspect.